File: docs/examples/rust/src/memory.rs
use libmem::*;

fn main() {
    let some_var : i32 = 10;
    println!("[*] Value of 'some_var': {}", some_var);

    let read_some_var = LM_ReadMemory::<i32>(&some_var as *const i32 as lm_address_t).unwrap();
    println!("[*] Read Value of 'some_var': {}", read_some_var);

    LM_WriteMemory::<i32>(&some_var as *const i32 as lm_address_t, &1337).unwrap();
    println!("[*] Value of 'some_var' after writing: {}", some_var);
}

File: examples/c/memory.c
#include <libmem/libmem.h>

int
main()
{
	int some_var = 10;
	int read_some_var;
	int write_some_var = 1337;
	printf("[*] Value of 'some_var': %d\n", some_var);

	LM_ReadMemory((lm_address_t)&some_var, (lm_byte_t *)&read_some_var, sizeof(read_some_var));
	printf("[*] Read Value of 'some_var': %d\n", read_some_var);

	LM_WriteMemory((lm_address_t)&some_var, (lm_bytearr_t)&write_some_var, sizeof(write_some_var));
	printf("[*] Value of 'some_var' after writing: %d\n", some_var);

	return 0;
}
